[Tan Jia Rong] iP

Tan-Jia-Rong opened the pull request:
# Duke 


> ╭━━━╮╱╱╭╮
> ╰╮╭╮┃╱╱┃┃
> ╱┃┃┃┣╮╭┫┃╭┳━━╮
> ╱┃┃┃┃┃┃┃╰╯┫┃━┫
> ╭╯╰╯┃╰╯┃╭╮┫┃━┫
> ╰━━━┻━━┻╯╰┻━━╯

Too much tasks to do? Cant keep track of what you need to do? Duke is here to save the day! :superhero_man:

- text-based
- easy to learn
- It goes ~~fast~~ **ZOOOOM** :bullettrain_front:

All you need to do is,

1. Download it from [here](https://github.com/Tan-Jia-Rong/ip/releases/tag/A-Jar)
2. Open Terminal and navigate to where *duke.jar* is located
3. Run `java -jar duke.jar`
4. Add your tasks.
5. let it manage your tasks for you :wink:
## Features

- [x] Manage Tasks (ToDo, Deadline, Event)
- [x] Provide Command List
- [ ] Reminders

## Commands

Here are some basic commands:

```java
        // Return Task in TaskList
        list

        // Set selected task as completed
        mark index of task

        // Set selected task as not completed
        unmark index of task

        // Delete selected task
        delete index of task

        // Adds a Todo Task
        Todo description

        // Adds a Deadline Task
        Deadline description /by YYYY-MM-DD 24hr

        // Adds a Event Task
        Event description /at YYYY-MM-DD 24hr
```

##
If you are a java programmer, here's the `main` method to run it:

```java
    public static void main(String[] args) {
        Duke duke = new Duke(SAVE_LOCATION);
        duke.run();
    }
```

Commit:
Add Gradle support

Tan-Jia-Rong committed:
Added Level 1 Functionalities

Tan-Jia-Rong committed:
Added Level 2 Functionalities

Tan-Jia-Rong committed:
Added Level 3 Functionalities

Tan-Jia-Rong committed:
add indentation for mark and unmark

Tan-Jia-Rong committed:
Added ToDo func & clean code

Tan-Jia-Rong committed:
Added Deadline Functionality

Tan-Jia-Rong committed:
Added Event functionality

Tan-Jia-Rong committed:
Added Text-UI_test and fixed code

Tan-Jia-Rong committed:
Added input and expected Result

Tan-Jia-Rong committed:
Added Exceptions and Update Testing

Tan-Jia-Rong committed:
Added Delete Feature and Updated Testing

Tan-Jia-Rong committed:
Update indentation error

Tan-Jia-Rong committed:
Clean up code

Tan-Jia-Rong committed:
Update JavaDocs

Tan-Jia-Rong committed:
Add Level-7 Functionalities

Tan-Jia-Rong committed:
Add Level-8 Functionalities

Tan-Jia-Rong committed:
Merge Level-7 Functionalities

Tan-Jia-Rong committed:
Merge Level-8 Functionalities

Tan-Jia-Rong committed:
Refactor Code: Add Storage, Parser, TaskList, Ui

Tan-Jia-Rong committed:
Add packages

Tan-Jia-Rong committed:
Update Duke and UI

Tan-Jia-Rong committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

* 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip:
  Add Gradle support

Tan-Jia-Rong committed:
Set up Gradle with CheckStyle

Tan-Jia-Rong committed:
Update README.md

Tan-Jia-Rong committed:
Add JUnit tests

Tan-Jia-Rong committed:
Update build.gradle path

Tan-Jia-Rong committed:
Update JavaDocs

Tan-Jia-Rong committed:
Comply to coding standards

Tan-Jia-Rong committed:
Add Level-9 Functionalities

Tan-Jia-Rong committed:
Merge branch 'branch-A-JavaDoc'

Tan-Jia-Rong committed:
Merge branch 'branch-A-CodingStandard'

Tan-Jia-Rong committed:
Merge branch 'branch-Level-9'

Tan-Jia-Rong committed:
Fix merge conflicts

Tan-Jia-Rong committed:
Update error in Ui

Guanzhou03 commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.duke.java` file, right-click it, and choose `Run duke.duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Good that package name is lowercase

Tan-Jia-Rong replied:
Was thinking that I shouldn't write the duke.duke.java because it looks weird 😅  Guess ill edit it!
Thank you for your review!

Tan-Jia-Rong replied:
Thanks! 😄

Guanzhou03 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     */
+    public void execute(String command, String description, TaskList tasks) {
+        try {
+            if (command.equalsIgnoreCase("list")) {
+                ui.listTasks(tasks);
+            } else if (command.equalsIgnoreCase("mark")) {
+                int index = Integer.parseInt(description.substring(1)) - 1;
+                tasks.mark(index);
+            } else if (command.equalsIgnoreCase("unmark")) {
+                int index = Integer.parseInt(description.substring(1)) - 1;
+                tasks.unmark(index);
+            } else if (command.equalsIgnoreCase("delete")) {
+                int index = Integer.parseInt(description.substring(1)) - 1;
+                tasks.delete(index);
+            } else if (command.equalsIgnoreCase("todo")) {
+                if (description.isEmpty()) {
+                    throw new DukeEmptyToDoException();
+                }
+                tasks.add(new ToDo(description));
+            } else if (command.equalsIgnoreCase("deadline")) {
+                if (description.isEmpty()) {
+                    throw new DukeEmptyDeadlineException();
+                }
+                String[] input = description.split(" /by ");
+                tasks.add(new Deadline(input[0], input[1]));
+            } else if (command.equalsIgnoreCase("event")) {
+                if (description.isEmpty()) {
+                    throw new DukeEmptyEventException();
+                }
+                String[] input = description.split(" /at ");
+                tasks.add(new Event(input[0], input[1]));
+            } else if (command.equalsIgnoreCase("find")) {
+                tasks.findTasks(description);
+            } else if (command.equalsIgnoreCase("help")) {
+                ui.helpMessage();
+            } else {
+                throw new DukeInvalidCommandException();
+            }
```
Can consider using switch case to reduce the number of lines and improve readability

Tan-Jia-Rong replied:
It is a good idea. I will implement your suggestion. Thank you for your review!

Guanzhou03 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,117 @@
+    /**
+     * Sets Task to be done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
```
Good use of verbs for naming.

Tan-Jia-Rong replied:
Thanks 😄

Guanzhou03 commented on src/test/java/duke/task/TaskListTest.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    @Test
+    public void addTask_normalInput_worksNormally() {
```
I think a test already tests if something works normally, perhaps the naming can omit the worksNormally for more concise code

Tan-Jia-Rong replied:
It is a good idea. I will implement your suggestion. Thank you for your review!

Guanzhou03 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+ */
+public class Duke {
+    // Location of save file
+    private static final String SAVE_LOCATION = "./data/data.txt";
```
Good use of capitalisation for constants

Tan-Jia-Rong replied:
Thanks ❤️

Tan-Jia-Rong committed:
Improve Test Code Quality

Tan-Jia-Rong committed:
Improve Code Quality

teoyuqi commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,178 @@
+    /**
+     * Prints Greetings Message.
+     */
+    public void greetings() {
```
I believe the course's [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#naming) requires us to use verbs for methods.

> Names representing methods must be verbs and written in camelCase.
> ```getName(), computeTotalWidth()```

Perhaps we could use something like `sendGreeting()` or `greet()`?

This issue is repeated in subsequent lines of `Ui.java` as well:
 * `farewell()` -> `sendFarewell` or `bidFarewell`?
 * `helpMessage()` -> `sendHelpMessage` or `showHelpMessage`?

Tan-Jia-Rong replied:
Thank you I have since updated my method Names ❤️

teoyuqi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,95 @@
+            int index;
+            String[] input;
+            command = command.toLowerCase();
+            switch (command) {
```
Great job in keeping `switch` and `case` on the same level of indentation!

Tan-Jia-Rong replied:
Yes! 👍

teoyuqi reviewed (commented):
Your code and documentation is very readable and clear to readers. Good job! Other than a few minor naming issues in `Ui.java`, the code looks good to me! 👍🏻

Tan-Jia-Rong committed:
Refactor Code

Tan-Jia-Rong committed:
Adhere to Coding Standards

Tan-Jia-Rong committed:
Add basic JavaFx functionalities

Tan-Jia-Rong committed:
Improve JavaFx Gui

Tan-Jia-Rong committed:
Clean up code

Tan-Jia-Rong committed:
Adhere to coding standard

Tan-Jia-Rong committed:
Refactor Code

Tan-Jia-Rong committed:
Clean up code

shamooose reviewed (commented):
Good organisation of your code!

Tan-Jia-Rong committed:
Improve Code Quality

Enhance code abstraction in Parser and Storage class by applying SLAP.
Restructured code to utilise happy path where possible.
Added named constant where applicable to avoid magic numbers.
Removed complicated expressions and stored them in variables.
Created multiple exceptions to minimise code duplication.

Tan-Jia-Rong committed:
Add Assertions

Included Assertion statement on various point in the code to check for
validity of program state during runtime.

Tan-Jia-Rong committed:
Merge pull request #1 from Tan-Jia-Rong/branch-A-CodeQuality

Improve Code Quality

Tan-Jia-Rong committed:
Merge branch 'master' into branch-A-Assertions

Tan-Jia-Rong committed:
Update assertion

Added more assertion is various points on the program.

Tan-Jia-Rong committed:
Update .gitignore to include data.txt.

Tan-Jia-Rong committed:
Fix Error in .gitignore

Tan-Jia-Rong committed:
Merge pull request #2 from Tan-Jia-Rong/branch-A-Assertions

Add Assertions

Tan-Jia-Rong committed:
Add BatchTypeDelete

Tan-Jia-Rong committed:
Add Batch delete by description functionalities

Tan-Jia-Rong committed:
Merge pull request #3 from Tan-Jia-Rong/branch-C-MassOps

Add MassOps functionalities.

Tan-Jia-Rong committed:
Merge branch 'master' into branch-Level-10
Branch for progress dashboard purposes

Tan-Jia-Rong committed:
Merge branch 'branch-Level-10'
Branch for iP dashboard purposes

Tan-Jia-Rong committed:
Update UI

Tan-Jia-Rong committed:
Add Credit

Tan-Jia-Rong committed:
Update testcases

Tan-Jia-Rong committed:
Update javafx

Tan-Jia-Rong committed:
Fixed Lag issues with JavaFx

JavaFx was laggy when using Text instead of Label.
Swapped back to Label for better performance.

Tan-Jia-Rong committed:
Adhere to coding standards

Tan-Jia-Rong committed:
Update README.md

Tan-Jia-Rong committed:
Update Ui

Tan-Jia-Rong committed:
Update README.md

Tan-Jia-Rong committed:
Update README.md

Tan-Jia-Rong committed:
Update README.md
